### PR TITLE
Support for ARM/Raspberry pi

### PR DESCRIPTION
**Pull request by natcl**
Arm architectures do not have the dmidecode command.  On the Raspberry pi the serial can be obtained with `cat /proc/cpuinfo | grep Serial`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -71,9 +71,14 @@ var serialNumber = function (cb, cmdPrefix) {
 		cmd = 'system_profiler SPHardwareDataType | grep ';
 		break;
 
-	case 'linux':
 	case 'freebsd':
-		cmd = 'dmidecode -t system | grep ';
+	case 'linux':
```
Comment (es128) [inline index.js:74]: Why did you change the order of linux/freebsd?

File: index.js [hunk 2]
```diff
@@ -71,9 +71,14 @@ var serialNumber = function (cb, cmdPrefix) {
...
+		if (process.arch === 'arm') {
+			cmd = 'cat /proc/cpuinfo | grep ';
```
Comment (es128) [inline index.js:77]: Just want to confirm - if this ends up being `grep UUID` will a valid response still be returned on ARM?

File: index.js (hunk 1 above)
Comment (natcl) [inline index.js:74]: Because I did not want to duplicate code in the freebsd section.

Comment (es128) [inline index.js:74]: There is no separate freebsd section. This is the syntax for giving two cases the same logic block.

File: index.js (hunk 2 above)
Comment (natcl) [inline index.js:77]: Indeed it fails in UUID mode, I'm not familiar with uuids, what would be a good solution ?

Comment (es128) [inline index.js:77]: `vals[1] = 'Serial'`

Comment (natcl) [inline index.js:77]: Thanks, did some corrections to the patch

**Comment by es128** [comment]
Looks good, thanks!
